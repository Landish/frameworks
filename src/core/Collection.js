class Collection {
  constructor(data = []) {
    this.data = data;
    this.count = data.length;
  }

  has(item) {
    return this.data.indexOf(item) > -1;
  }

  add(item) {
    if (!this.has(item)) {
      this.count = this.count + 1;
      this.data.push(item);
    }
    return this;
  }

  remove(item) {
    if (this.has(item)) {
      this.data.splice(this.data.indexOf(item), 1);
      this.count = this.count - 1;
    }
  }

  removeFirst() {
    this.data.shift();
    this.count = this.count - 1;
  }

  size() {
    return this.count;
  }
}

export default Collection;
