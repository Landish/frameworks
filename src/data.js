export default {
    "frameworks": [
        {
            "name": "Bootstrap 3",
            "slug": "bootstrap",
            "description": "The most popular HTML, CSS, and JS framework for developing responsive, mobile first projects on the web",
            "url": "http://getbootstrap.com",
            "documentation": "http://getbootstrap.com/getting-started/",
            "github": "https://github.com/twbs/bootstrap",
            "icons" : {
                "name" : "Glyphicons",
                "url" : "http://getbootstrap.com/components/#glyphicons-glyphs"
            },
            "css": "LESS",
            "js": "http://getbootstrap.com/javascript/",
            "stats": {
                "stars": "104K+",
                "forks": "47K+"
            }
        },
        {
            "name": "Semantic UI",
            "slug": "semantic-ui",
            "description": "Semantic is a UI component framework based around useful principles from natural language",
            "url": "http://semantic-ui.com/",
            "documentation": "http://semantic-ui.com/introduction/getting-started.html",
            "github": "https://github.com/Semantic-Org/Semantic-UI",
            "icons" : {
                "name" : "Font Awesome",
                "url" : "http://fontawesome.io/"
            },
            "css": "LESS",
            "js": "http://semantic-ui.com/behaviors/api.html",
            "stats": {
                "stars": "30K+",
                "forks": "3.4K+"
            }
        },
        {
            "name": "UIKit",
            "slug": "uikit",
            "description": "A lightweight and modular front-end framework for developing fast and powerful web interfaces",
            "url": "https://getuikit.com/",
            "documentation": "https://getuikit.com/docs/documentation_get-started.html",
            "github": "https://github.com/uikit/uikit",
            "icons" : {
                "name" : "Font Awesome",
                "url" : "http://fontawesome.io/"
            },
            "css": "LESS",
            "js": "https://getuikit.com/docs/documentation_javascript.html",
            "stats": {
                "stars": "8K+",
                "forks": "1.5K+"
            }
        },
        {
            "name": "Foundation 6",
            "slug": "foundation",
            "install": "foundation-sites",
            "description": "The most advanced responsive front-end framework in the world",
            "url": "http://foundation.zurb.com/",
            "documentation": "http://foundation.zurb.com/sites/docs/",
            "github": "https://github.com/zurb/foundation-sites",
            "icons" : {
                "name" : "Foundation Icons",
                "url" : "http://foundation.zurb.com/icon-fonts.html"
            },
            "css": "SASS",
            "js": "https://foundation.zurb.com/sites/docs/javascript.html",
            "stats": {
                "stars": "24K+",
                "forks": "5K+"
            }
        },
        {
            "name": "Bulma",
            "slug": "bulma",
            "description": "A modern CSS framework based on Flexbox",
            "url": "http://bulma.io/",
            "documentation": "http://bulma.io/documentation/overview/start/",
            "github": "https://github.com/jgthms/bulma",
            "icons" : {
                "name" : "Font Awesome (Optional)",
                "url" : "http://fontawesome.io/"
            },
            "css": "SASS",
            "js": false,
            "stats": {
                "stars": "10K+",
                "forks": "500+"
            }
        }
    ],
    "components": [
        {
            "name": "Alert",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#alerts",
                "uikit": "https://getuikit.com/docs/alert.html",
                "semantic-ui": "http://semantic-ui.com/collections/message.html",
                "foundation" : "http://foundation.zurb.com/sites/docs/callout.html",
                "bulma": "http://bulma.io/documentation/elements/notification/"
            }
        },
        {
            "name": "Badge",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#badges",
                "semantic-ui": "http://semantic-ui.com/elements/label.html#circular",
                "uikit": "https://getuikit.com/docs/badge.html",
                "foundation": "http://foundation.zurb.com/sites/docs/badge.html",
                "bulma": "http://bulma.io/documentation/elements/tag/"
            }
        },
        {
            "name": "Button",
            "supports": {
                "bootstrap": "http://getbootstrap.com/css/#buttons",
                "semantic-ui": "http://semantic-ui.com/elements/button.html",
                "uikit": "https://getuikit.com/docs/button.html",
                "foundation": "http://foundation.zurb.com/sites/docs/button.html",
                "bulma": "http://bulma.io/documentation/elements/button/"
            }
        },
        {
            "name": "Button Groups",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#btn-groups",
                "semantic-ui": "http://semantic-ui.com/elements/button.html#buttons",
                "uikit": "https://getuikit.com/docs/button.html#button-group",
                "foundation": "http://foundation.zurb.com/sites/docs/button-group.html",
                "bulma": "http://bulma.io/documentation/elements/button/"
            }
        },
        {
            "name": "Breadcrumbs",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#breadcrumbs",
                "semantic-ui": "http://semantic-ui.com/collections/breadcrumb.html",
                "foundation": "http://foundation.zurb.com/sites/docs/breadcrumbs.html",
                "uikit": "https://getuikit.com/docs/breadcrumb.html"
            }
        },
        {
            "name": "Card",
            "supports": {
                "semantic-ui": "http://semantic-ui.com/views/card.html",
                "bulma": "http://bulma.io/documentation/components/card/"
            }
        },
        {
            "name": "Dropdown",
            "supports": {
                "bootstrap": "http://getbootstrap.com/javascript/#dropdowns",
                "semantic-ui": "http://semantic-ui.com/modules/dropdown.html",
                "uikit": "https://getuikit.com/docs/dropdown.html",
                "foundation": "http://foundation.zurb.com/sites/docs/dropdown-menu.html"
            }
        },
        {
            "name": "Flex Grid",
            "supports": {
                "semantic-ui": "http://semantic-ui.com/collections/grid.html",
                "uikit": "https://getuikit.com/docs/flex.html",
                "foundation": "https://foundation.zurb.com/sites/docs/flex-grid.html",
                "bulma": "http://bulma.io/documentation/grid/columns/"
            }
        },
        {
            "name": "Grid",
            "supports": {
                "bootstrap": "http://getbootstrap.com/css/#grid",
                "semantic-ui": "http://semantic-ui.com/collections/grid.html",
                "uikit": "https://getuikit.com/docs/grid.html",
                "foundation": "https://foundation.zurb.com/sites/docs/grid.html",
                "bulma": "http://bulma.io/documentation/grid/columns/"
            }
        },
        {
            "name": "Form",
            "supports": {
                "bootstrap": "http://getbootstrap.com/css/#forms",
                "semantic-ui": "http://semantic-ui.com/collections/form.html",
                "uikit": "https://getuikit.com/docs/form.html",
                "foundation": "https://foundation.zurb.com/sites/docs/forms.html",
                "bulma": "http://bulma.io/documentation/elements/form/"
            }
        },
        {
            "name": "Icons",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#glyphicons-glyphs",
                "semantic-ui": "http://semantic-ui.com/elements/icon.html",
                "uikit": "https://getuikit.com/docs/icon.html",
                "bulma": "http://bulma.io/documentation/elements/icon/"
            }
        },
        {
            "name": "Label",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#labels",
                "semantic-ui": "http://semantic-ui.com/elements/label.html",
                "uikit": "https://getuikit.com/docs/badge.html",
                "foundation": "http://foundation.zurb.com/sites/docs/label.html",
                "bulma": "http://bulma.io/documentation/elements/tag/"
            }
        },
        {
            "name": "List",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#list-group",
                "semantic-ui": "http://semantic-ui.com/elements/list.html",
                "uikit": "https://getuikit.com/docs/list.html"
            }
        },
        {
            "name": "Media Object",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#media",
                "foundation": "http://foundation.zurb.com/sites/docs/media-object.html",
                "bulma": "http://bulma.io/documentation/components/media-object/"
            }
        },
        {
            "name": "Modal",
            "supports": {
                "bootstrap": "http://getbootstrap.com/javascript/#modals",
                "semantic-ui": "http://semantic-ui.com/modules/modal.html",
                "uikit": "https://getuikit.com/docs/modal.html",
                "foundation": "http://foundation.zurb.com/sites/docs/reveal.html",
                "bulma": "http://bulma.io/documentation/components/modal/"
            }
        },
        {
            "name": "Navbar / Navigation",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#navbar",
                "semantic-ui": "http://semantic-ui.com/collections/menu.html",
                "uikit": "https://getuikit.com/docs/navbar.html",
                "bulma": "http://bulma.io/documentation/components/nav/"
            }
        },
        {
            "name": "Panel",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#panels",
                "uikit": "https://getuikit.com/docs/panel.html",
                "bulma": "http://bulma.io/documentation/components/panel/"
            }
        },
        {
            "name": "Pagination",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#pagination",
                "semantic-ui": "http://semantic-ui.com/collections/menu.html#pagination",
                "uikit": "https://getuikit.com/docs/pagination.html",
                "foundation": "http://foundation.zurb.com/sites/docs/pagination.html",
                "bulma": "http://bulma.io/documentation/components/pagination/"
            }
        },
        {
            "name": "Progress",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#progress",
                "semantic-ui": "http://semantic-ui.com/modules/progress.html",
                "uikit": "https://getuikit.com/docs/progress.html",
                "foundation": "http://foundation.zurb.com/sites/docs/progress-bar.html",
                "bulma": "http://bulma.io/documentation/elements/progress/"
            }
        },
        {
            "name": "Table",
            "supports": {
                "bootstrap": "http://getbootstrap.com/css/#tables",
                "semantic-ui": "http://semantic-ui.com/collections/table.html",
                "uikit": "https://getuikit.com/docs/table.html",
                "foundation": "http://foundation.zurb.com/sites/docs/table.html",
                "bulma": "http://bulma.io/documentation/elements/table/"
            }
        },
        {
            "name": "Tabs",
            "supports": {
                "bootstrap": "http://getbootstrap.com/javascript/#tabs",
                "semantic-ui": "http://semantic-ui.com/modules/tab.html",
                "uikit": "https://getuikit.com/docs/tab.html",
                "bulma": "http://bulma.io/documentation/components/tabs/"
            }
        },
        {
            "name": "Thumbnail",
            "supports": {
                "bootstrap": "http://getbootstrap.com/components/#thumbnails",
                "semantic-ui": "http://semantic-ui.com/elements/image.html",
                "uikit": "https://getuikit.com/docs/thumbnail.html",
                "foundation": "https://foundation.zurb.com/sites/docs/thumbnail.html",
                "bulma": "http://bulma.io/documentation/elements/image/"
            }
        },
        {
            "name": "Tooltip",
            "supports": {
                "bootstrap": "http://getbootstrap.com/javascript/#tooltips",
                "semantic-ui": "http://semantic-ui.com/modules/popup.html#tooltip",
                "foundation": "http://foundation.zurb.com/sites/docs/tooltip.html",
                "uikit": "https://getuikit.com/docs/tooltip.html"
            }
        }
    ]
}