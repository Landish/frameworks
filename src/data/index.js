import { AmazeUI } from './frameworks/AmazeUI';
import { BlazeCSS } from './frameworks/BlazeCSS';
import { Bootstrap3, Bootstrap4 } from './frameworks/Bootstrap';
import { Bulma } from './frameworks/Bulma';
import { Foundation5, Foundation6 } from './frameworks/Foundation';
import { Ink } from './frameworks/Ink';
import { Kube } from './frameworks/Kube';
import { Materialize } from './frameworks/Materialize';
import { UIkit2, UIkit3 } from './frameworks/UIkit';
import { Pure } from './frameworks/Pure';
import { Primer } from './frameworks/Primer';
import { SemanticUI } from './frameworks/SemanticUI';
import { Spectre } from './frameworks/Spectre';

import Accordion from './components/Accordion';
import Alert from './components/Alert';
import Avatar from './components/Avatar';
import Badge from './components/Badge';
import Breadcrumbs from './components/Breadcrumbs';
import Button from './components/Button';
import Card from './components/Card';
import Dropdown from './components/Dropdown';
import FlexGrid from './components/FlexGrid';
import Form from './components/Form';
import Grid from './components/Grid';
import Icons from './components/Icons';
import Label from './components/Label';
import List from './components/List';
import Media from './components/Media';
import Menu from './components/Menu';
import Modal from './components/Modal';
import Navbar from './components/Navbar';
import OffCanvas from './components/OffCanvas';
import Pagination from './components/Pagination';
import Panel from './components/Panel';
import Popover from './components/Popover';
import Progress from './components/Progress';
import Table from './components/Table';
import Tabs from './components/Tabs';
import Thumbnail from './components/Thumbnail';
import Tooltip from './components/Tooltip';
import Typography from './components/Typography';
import Utilities from './components/Utilities';

export default {
  frameworks: [
    AmazeUI,
    BlazeCSS,
    Bootstrap3,
    Bootstrap4,
    Bulma,
    Foundation5,
    Foundation6,
    Ink,
    Kube,
    Materialize,
    UIkit2,
    UIkit3,
    Pure,
    Primer,
    SemanticUI,
    Spectre
  ],
  components: [
    Accordion,
    Alert,
    Avatar,
    Badge,
    Breadcrumbs,
    Button,
    Card,
    Dropdown,
    FlexGrid,
    Form,
    Grid,
    Icons,
    Label,
    List,
    Media,
    Menu,
    Modal,
    Navbar,
    OffCanvas,
    Pagination,
    Panel,
    Popover,
    Progress,
    Table,
    Tabs,
    Thumbnail,
    Tooltip,
    Typography,
    Utilities
  ]
};
