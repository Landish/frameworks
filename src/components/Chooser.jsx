import React from 'react';
import PropTypes from 'prop-types';
import Collection from '../core/Collection';
import Icon from './Icon';

const Chooser = ({ frameworks, visible, onFrameworkSelect }) => {
  return (
    <div className="section has-text-centered" id="chooser">
      <div className="container">
        <h2 className="title is-5">
          Select <strong>at least 2</strong> and <strong>maximum 4</strong>{' '}
          front-end frameworks and compare them to each other.
        </h2>
        <div className="notification">
          {frameworks.map(framework => {
            return (
              <button
                key={framework.slug}
                onClick={() => {
                  onFrameworkSelect(framework.slug);
                }}
                className={`${visible.has(framework.slug)
                  ? ''
                  : 'is-outlined '}button is-checkbox is-info`}
              >
                <Icon
                  name={visible.has(framework.slug) ? 'check-square' : 'square'}
                />
                <span>
                  {framework.name}
                </span>
              </button>
            );
          })}
        </div>
      </div>
    </div>
  );
};

Chooser.propTypes = {
  frameworks: PropTypes.array.isRequired,
  visible: PropTypes.instanceOf(Collection).isRequired,
  onFrameworkSelect: PropTypes.func.isRequired
};

export default Chooser;
