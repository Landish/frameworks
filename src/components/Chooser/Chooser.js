import React from 'react';
import { array, func, number, instanceOf } from 'prop-types';
import SelectButton from '../SelectButton';
import Collection from '../../core/Collection';

const Chooser = ({ frameworks, minimum, maximum, visible, onFrameworkSelect }) => {
  return (
    <div className="section has-text-centered">
      <div className="container is-fullhd">
        <h2 className="subtitle is-5">
          Select <strong>at least {minimum}</strong> and <strong>maximum {maximum}</strong>{' '}
          front-end frameworks and compare them to each other.
        </h2>
        <div className="notification">
          {frameworks.map(framework => {
            return (
              <SelectButton
                key={framework.slug}
                isNew={framework.isNew}
                isSelected={visible.has(framework.slug)}
                framework={framework}
                onClick={() => {
                  onFrameworkSelect(framework.slug);
                }}
              />
            );
          })}
        </div>
      </div>
    </div>
  );
};

Chooser.propTypes = {
  frameworks: array.isRequired,
  visible: instanceOf(Collection).isRequired,
  onFrameworkSelect: func.isRequired,
  minimum: number,
  maximum: number
};

Chooser.defaultProps = {
  minimum: 2,
  maximum: 4
};

export default Chooser;
