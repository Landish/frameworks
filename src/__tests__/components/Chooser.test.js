import React from 'react';
import Chooser from '../../components/Chooser';
import Collection from '../../core/Collection';
import data from '../../data';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const visible = new Collection(['bulma']);
  const tree = renderer
    .create(
      <Chooser
        frameworks={data.frameworks}
        visible={visible}
        onFrameworkSelect={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
