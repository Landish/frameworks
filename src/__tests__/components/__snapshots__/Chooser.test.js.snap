// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="section has-text-centered"
  id="chooser"
>
  <div
    className="container"
  >
    <h2
      className="title is-5"
    >
      Select 
      <strong>
        at least 2
      </strong>
       and 
      <strong>
        maximum 4
      </strong>
       
      front-end frameworks and compare them to each other.
    </h2>
    <div
      className="notification"
    >
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          Bootstrap 3
        </span>
      </button>
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          Bootstrap 4 (Alpha)
        </span>
      </button>
      <button
        className="button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-check-square"
          />
        </span>
        <span>
          Bulma
        </span>
      </button>
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          Foundation 6
        </span>
      </button>
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          Foundation 5
        </span>
      </button>
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          Ink Interface Kit
        </span>
      </button>
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          Kube
        </span>
      </button>
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          Materialize
        </span>
      </button>
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          UIkit 2
        </span>
      </button>
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          UIkit 3 (Beta)
        </span>
      </button>
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          Pure.css
        </span>
      </button>
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          Primer
        </span>
      </button>
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          Semantic UI
        </span>
      </button>
      <button
        className="is-outlined button is-checkbox is-info"
        onClick={[Function]}
      >
        <span
          className="icon is-small"
        >
          <i
            className="fa fa-square"
          />
        </span>
        <span>
          Spectre.css
        </span>
      </button>
    </div>
  </div>
</div>
`;
